Modules/posixmodule.c

```diff
@@ -10035,6 +10040,205 @@ os_times_impl(PyObject *module)
+    int result;
+    Py_BEGIN_ALLOW_THREADS
+    result = timerfd_gettime(fd, &curr_value);
+    Py_END_ALLOW_THREADS
+    if (result == -1) {
+        return PyErr_SetFromErrno(PyExc_OSError);
+    }
+    return build_itimerspec(&curr_value);
+}
+
+/*[clinic input]
+os.timerfd_settime
+
+    fd: fildes
+        a file descriptor for timer.
+    flags: int
+        0 or  a bit mask of TFD_TIMER_ABSTIME or TFD_TIMER_CANCEL_ON_SET.
+    it_interval: double = 0.0
+        If TFD_TIMER_ABSTIME is not specified, it_interval is relative time.
+        If TFD_TIMER_ABSTIME is specified, it_interval is absolute time and epoch time in time_t.
+        If TFD_TIMER_CANCEL_ON_SET  and TFD_TIMER_ABSTIME are specifed as *flags* and time.CLOCK_REALTIME
+        is specified as *clockid* of timerfd_create, a discontinuous change of system clock will make to
+        abort reading from a file descriptor with ECANCELED error.
+        it_interval is interval for timer in seconds.
+    it_value: double = 0.0
+        it_value is initial expiration timing in seconds. It could be absolute time or relative time
+        based on TFD_TIMER_ABSTIME flag.
+    /
+
+Set timerfd value in seconds.
+[clinic start generated code]*/
+
+static PyObject *
+os_timerfd_settime_impl(PyObject *module, int fd, int flags,
+                        double it_interval, double it_value)
+/*[clinic end generated code: output=24030b5b2c5b539c input=915cd54d9ef8a889]*/
+{
+    struct itimerspec new_value;
+    struct itimerspec old_value;
+    int result;
+    new_value.it_interval.tv_sec = (time_t)it_interval;
+    new_value.it_interval.tv_nsec = EXTRACT_NSEC(it_interval);
+    new_value.it_value.tv_sec = (time_t)it_value;
+    new_value.it_value.tv_nsec = EXTRACT_NSEC(it_value);
+    Py_BEGIN_ALLOW_THREADS
+    result = timerfd_settime(fd, flags, &new_value, &old_value);
+    Py_END_ALLOW_THREADS
+    if (result == -1) {
+        return PyErr_SetFromErrno(PyExc_OSError);
+    }
+    return build_itimerspec(&old_value);
+}
+
+/*[clinic input]
+os.timerfd_gettime_ns
+
+    fd: fildes
+        a file descriptor for timer.
+    /
+
+Return timerfd value in nanoseconds.
+[clinic start generated code]*/
+
+static PyObject *
+os_timerfd_gettime_ns_impl(PyObject *module, int fd)
+/*[clinic end generated code: output=580633a4465f39fe input=731cd1818f472501]*/
+{
+    struct itimerspec curr_value;
+    int result;
+    Py_BEGIN_ALLOW_THREADS
+    result = timerfd_gettime(fd, &curr_value);
+    Py_END_ALLOW_THREADS
+    if (result == -1) {
+        return PyErr_SetFromErrno(PyExc_OSError);
+    }
+    return build_itimerspec_ns(&curr_value);
+}
+
+/*[clinic input]
+os.timerfd_settime_ns
+
+    fd: fildes
+        a file descriptor for timer.
+    flags: int
+        similar to timerfd_settime
+    it_interval_ns: long_long = 0
+        similar to timerfd_settime except for in nanoseconds.
+    it_value_ns: long_long = 0
+        similar to timerfd_settime except for in nanoseconds.
```

Comment:
See comments on ``timerfd_settime``. Please also copy the full docstrings here rather than saying read X instead.